src/metadata.rs

```diff
@@ -258,18 +261,24 @@ impl<'config> MetadataFetcher for CargoInternalsMetadataFetcher<'config> {
         features.insert(feature.clone(), features_or_dependencies.clone());
       }
 
+      let git_root = self.cargo_config.git_path().into_path_unlocked();
```

Comment:
This comment is closely tied into my comment on the `Package` struct.

Is it at all possible to do this without the `cargo_config`? I ask because we might need an alternative to this for the `CargoSubcommandMetadataFetcher` if we keep the `checkout_root` arg in `Package` -- ideally we can do this entirely without using cargo context and only using `Metadata`

Reply:
Yeah, this is the only reason I needed to edit `Package`. Ultimately, the reason you need `cargo_config` is because otherwise I'm re-writing the logic that Cargo uses for determining the checkout directory without Cargo's input.

What would potentially work instead is something where I shell out to the system `cargo` and that echos back the `git_root`, but I'm hesitant to try and re-engineer Cargo.